cubi_tk/sodar/ingest.py

```diff
@@ -0,0 +1,297 @@
+        )
+        parser.add_argument(
+            "sources", help="One or multiple files/directories to ingest.", nargs="+"
+        )
+        parser.add_argument("destination", help="UUID or iRODS path of SODAR landing zone.")
+
+    @classmethod
+    def run(
+        cls, args, _parser: argparse.ArgumentParser, _subparser: argparse.ArgumentParser
+    ) -> typing.Optional[int]:
+        """Entry point into the command."""
+        return cls(args).execute()
+
+    def execute(self):
+        """Execute ingest."""
+        # Retrieve iRODS path if destination is UUID
+        if is_uuid(self.args.destination):
+            try:
+                lz_info = api.landingzone.retrieve(
+                    sodar_url=self.args.sodar_url,
+                    sodar_api_token=self.args.sodar_api_token,
+                    landingzone_uuid=self.args.destination,
+                )
+            except Exception as e:  # pragma: no cover
+                logger.error("Failed to retrieve landing zone information.")
+                logger.exception(e)
+                sys.exit(1)
+
+            # TODO: Replace with status_locked check once implemented in sodar_cli
+            if lz_info.status in ["ACTIVE", "FAILED"]:
+                self.lz_irods_path = lz_info.irods_path
+                logger.info(f"Target iRods path: {self.lz_irods_path}")
+            else:
+                logger.error("Target landing zone is not ACTIVE.")
+                sys.exit(1)
+        else:
+            self.lz_irods_path = self.args.destination  # pragma: no cover
+
+        # Build file list
+        source_paths = self.build_file_list()
+        if len(source_paths) == 0:
+            logger.info("Nothing to do. Quitting.")
+            sys.exit(0)
+
+        # Initiate iRODS session
+        irods_session = iRODSCommon().session
+
+        # Query target collection
+        logger.info("Querying landing zone collections…")
+        collections = []
+        try:
+            with irods_session as i:
+                coll = i.collections.get(self.lz_irods_path)
+                for c in coll.subcollections:
+                    collections.append(c.name)
+        except Exception as e:  # pragma: no cover
+            logger.error(
+                f"Failed to query landing zone collections: {iRODSCommon().get_irods_error(e)}"
+            )
+            sys.exit(1)
+
+        # Query user for target sub-collection
+        if not collections:
+            self.target_coll = self.lz_irods_path
+            logger.info("No subcollections found. Moving on.")
+        elif self.args.collection is None:
+            user_input = ""
+            input_valid = False
+            input_message = "####################\nPlease choose target collection:\n"
+            for index, item in enumerate(collections):
+                input_message += f"{index+1}) {item}\n"
+            input_message += "Select by number: "
```

Comment:
This part sort of ignores the "--yes" flag (in that it still asks for user input even with --yes), maybe add a note to the cmd-line flag help strings about this?
Could also add a check_args function that gives error if --yes is set without a set collection

Reply:
help text added